agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/exporter/LoggerExporter.java

```diff
@@ -0,0 +1,359 @@
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.util.Collection;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LoggerExporter implements LogExporter {
+
+  private static final Logger logger = LoggerFactory.getLogger(LoggerExporter.class);
+
+  private static final AttributeKey<String> AI_OPERATION_NAME_KEY =
+      AttributeKey.stringKey("applicationinsights.internal.operation_name");
+
+  private static final OperationLogger exportingLogLogger =
+      new OperationLogger(Exporter.class, "Exporting log");
+
+  private final TelemetryClient telemetryClient;
+  private final AtomicBoolean stopped = new AtomicBoolean();
+
+  public LoggerExporter(TelemetryClient telemetryClient) {
+    this.telemetryClient = telemetryClient;
+  }
+
+  @Override
+  public CompletableResultCode export(Collection<LogData> logs) {
+    if (CoreUtils.isNullOrEmpty(TelemetryClient.getActive().getInstrumentationKey())) {
+      logger.debug("Instrumentation key is null or empty. Fail to export logs.");
+      return CompletableResultCode.ofFailure();
+    }
+
+    if (stopped.get()) {
+      logger.debug("LoggerExporter has been stopped. Fail to export logs.");
+      return CompletableResultCode.ofFailure();
+    }
+
+    boolean failure = false;
+    for (LogData log : logs) {
+      try {
+        internalExport(log);
+        exportingLogLogger.recordSuccess();
+      } catch (Throwable t) {
+        exportingLogLogger.recordFailure(t.getMessage(), t);
+        failure = true;
+      }
+    }
+
+    return failure ? CompletableResultCode.ofFailure() : CompletableResultCode.ofSuccess();
+  }
+
+  @Override
+  public CompletableResultCode flush() {
+    return CompletableResultCode.ofSuccess();
+  }
+
+  @Override
+  public CompletableResultCode shutdown() {
+    stopped.set(true);
+    return CompletableResultCode.ofSuccess();
+  }
+
+  private void internalExport(LogData log) {
+    int severity = log.getSeverity().getSeverityNumber();
+    int threshold = getThreshold().getSeverityNumber();
+    if (severity < threshold) {
+      return;
+    }
+
+    logger.debug("exporting log: {}", log); // do we need to log this, will that be too much?
+    String stack = log.getAttributes().get(SemanticAttributes.EXCEPTION_STACKTRACE);
+    if (stack == null) {
+      trackMessage(log);
+    } else {
+      trackMessageAsException(log, stack);
+    }
+  }
+
+  private void trackMessage(LogData log) {
+    MessageTelemetryBuilder telemetryBuilder = telemetryClient.newMessageTelemetryBuilder();
+
+    Attributes attributes = log.getAttributes();
+
+    // set standard properties
+    telemetryBuilder.setTime(FormattedTime.offSetDateTimeFromEpochNanos(log.getEpochNanos()));
+    setOperationTags(telemetryBuilder, log);
+    setSampleRate(telemetryBuilder, log);
```

Comment:
Is this defined anywhere in common schema?

Reply:
not defined yet currently.. but it should be defined though since we send this in all telemetry..